app/controllers/application_controller.rb

```diff
@@ -220,6 +220,15 @@ def render_error(msg, status)
     render_403
   end
 
+  def render_extensible(page_view = action_name)
+    @page = ::Pages::Manager.find_page(controller_name, action_name)
+    if @page && @page.view
+      render(@page.view)
+    else
+      raise ::ForemanException.new(N_("Page %s has no view assigned") % @page.name)
```

Comment:
This is not the correct form, please follow this pattern: http://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions

And get rid of all `fail` methods in the patch, use this approach for all failure states that are presented to the user. Messages that go into logs do not need to be translated.